BaragonService/src/main/java/com/hubspot/baragon/service/elb/ApplicationLoadBalancer.java

```diff
@@ -29,23 +62,401 @@ public ApplicationLoadBalancer(Optional<ElbConfiguration> configuration,
+                                   Collection<LoadBalancer> loadBalancers) {
+    /*
+    - Check that load balancers, baragon agents, target groups are on same VPC
+    - Check that load balancers, targets are on same subnet (== AZ)
+    - Check that all baragon agents are associated with a target on target group
+    - Check that load balancers has listeners, rules to make talk to target group
+     */
+
+    if (configuration.isPresent() && configuration.get().isCheckForCorrectVpc()) {
+      guaranteeSameVPC(targetGroup, baragonAgents, loadBalancers);
+    }
+
+    guaranteeHasAllTargets(targetGroup, baragonAgents);
+    guaranteeAzEnabled(baragonAgents, loadBalancers);
+    //guaranteeListenersPresent(targetGroup, loadBalancers);
+  }
+
+  /**
+   * De-register any targets representing agents that are not known to the BaragonService,
+   * or which otherwise need to be removed.
+   *
+   * @param targetGroup TargetGroup to check for old agents
+   * @param agents Known agents, to be used as a reference sheet
+   */
+  private void deregisterIfOld(BaragonGroup baragonGroup, TargetGroup targetGroup, Collection<BaragonAgentMetadata> agents) {
+    Collection<TargetDescription> targets = targetsInTargetGroup(targetGroup);
+    Collection<TargetDescription> removeableTargets = listOldTargets(baragonGroup, targets, agents);
+
+    for (TargetDescription removeableTarget : removeableTargets) {
+      try {
+        if (isLastHealthyInstance(removeableTarget, targetGroup)
+            && configuration.isPresent()
+            && !configuration.get().isRemoveLastHealthyEnabled()) {
+          LOG.info("Will not de-register target {} because it is last healthy instance in {}", removeableTarget, targetGroup);
+        } else {
+          elbClient.deregisterTargets(new DeregisterTargetsRequest()
+              .withTargetGroupArn(targetGroup.getTargetGroupArn())
+              .withTargets(removeableTarget));
+          LOG.info("De-registered target {} from target group {}", removeableTarget, targetGroup);
+        }
+      } catch (AmazonClientException acexn) {
+        LOG.error("Could not de-register target {} from target group {} due to error",
+            removeableTarget, targetGroup, acexn);
+        exceptionNotifier.notify(acexn, ImmutableMap
+            .of("targetGroup", targetGroup.getTargetGroupName()));
+      }
+    }
+  }
+
+  /**
+   * When this method completes, the target group, the agents, and the loadBalancers are
+   * all on the same VPC.
+   * The target group, each of the agents, and each of the load balancers should think
+   * that they are on the same VPC, otherwise they won't be able to talk to each other.
+   *
+   *
+   * @param targetGroup Group - and consequently all targets - to check
+   * @param agents Agents to check
+   * @param loadBalancers Load balances to check
+   */
+  private void guaranteeSameVPC(TargetGroup targetGroup,
```

Comment:
we should return/raise something from this method if we find a mismatch in the vpc. If the vpc does not match we know ahead of time that any register reuqests will fail